[2023-06-24T10:57:14.946+0000] {taskinstance.py:1084} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: spark_airflow_dag_v3.spark_submit_task manual__2023-06-24T10:57:14.022095+00:00 [queued]>
[2023-06-24T10:57:14.954+0000] {taskinstance.py:1084} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: spark_airflow_dag_v3.spark_submit_task manual__2023-06-24T10:57:14.022095+00:00 [queued]>
[2023-06-24T10:57:14.954+0000] {taskinstance.py:1282} INFO - 
--------------------------------------------------------------------------------
[2023-06-24T10:57:14.955+0000] {taskinstance.py:1283} INFO - Starting attempt 1 of 1
[2023-06-24T10:57:14.955+0000] {taskinstance.py:1284} INFO - 
--------------------------------------------------------------------------------
[2023-06-24T10:57:14.967+0000] {taskinstance.py:1303} INFO - Executing <Task(SparkSubmitOperator): spark_submit_task> on 2023-06-24 10:57:14.022095+00:00
[2023-06-24T10:57:14.972+0000] {standard_task_runner.py:55} INFO - Started process 2345 to run task
[2023-06-24T10:57:14.975+0000] {standard_task_runner.py:82} INFO - Running: ['***', 'tasks', 'run', 'spark_***_dag_v3', 'spark_submit_task', 'manual__2023-06-24T10:57:14.022095+00:00', '--job-id', '63', '--raw', '--subdir', 'DAGS_FOLDER/spark_jobs_dag2.py', '--cfg-path', '/tmp/tmpb82r5sm4']
[2023-06-24T10:57:14.975+0000] {standard_task_runner.py:83} INFO - Job 63: Subtask spark_submit_task
[2023-06-24T10:57:14.987+0000] {warnings.py:109} WARNING - /home/***/.local/lib/python3.8/site-packages/***/settings.py:250: DeprecationWarning: The sql_alchemy_conn option in [core] has been moved to the sql_alchemy_conn option in [database] - the old setting has been used, but please update your config.
  SQL_ALCHEMY_CONN = conf.get("database", "SQL_ALCHEMY_CONN")

[2023-06-24T10:57:15.031+0000] {task_command.py:388} INFO - Running <TaskInstance: spark_airflow_dag_v3.spark_submit_task manual__2023-06-24T10:57:14.022095+00:00 [running]> on host 846f24a1d814
[2023-06-24T10:57:15.083+0000] {taskinstance.py:1509} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_OWNER=***
AIRFLOW_CTX_DAG_ID=spark_***_dag_v3
AIRFLOW_CTX_TASK_ID=spark_submit_task
AIRFLOW_CTX_EXECUTION_DATE=2023-06-24T10:57:14.022095+00:00
AIRFLOW_CTX_TRY_NUMBER=1
AIRFLOW_CTX_DAG_RUN_ID=manual__2023-06-24T10:57:14.022095+00:00
[2023-06-24T10:57:15.090+0000] {base.py:73} INFO - Using connection ID 'spark_test' for task execution.
[2023-06-24T10:57:15.090+0000] {taskinstance.py:1775} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 156, in execute
    self._hook = self._get_hook()
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 165, in _get_hook
    return SparkSubmitHook(
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 154, in __init__
    self._connection = self._resolve_connection()
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 211, in _resolve_connection
    raise RuntimeError(
RuntimeError: The `spark-home` extra is not allowed any more. Please make sure one of ['spark-submit', 'spark2-submit', 'spark3-submit'] is available on the PATH, and set `spark-binary` if needed.
[2023-06-24T10:57:15.097+0000] {taskinstance.py:1321} INFO - Marking task as FAILED. dag_id=spark_***_dag_v3, task_id=spark_submit_task, execution_date=20230624T105714, start_date=20230624T105714, end_date=20230624T105715
[2023-06-24T10:57:15.106+0000] {standard_task_runner.py:100} ERROR - Failed to execute job 63 for task spark_submit_task (The `spark-home` extra is not allowed any more. Please make sure one of ['spark-submit', 'spark2-submit', 'spark3-submit'] is available on the PATH, and set `spark-binary` if needed.; 2345)
[2023-06-24T10:57:15.146+0000] {local_task_job.py:212} INFO - Task exited with return code 1
[2023-06-24T10:57:15.159+0000] {taskinstance.py:2585} INFO - 0 downstream tasks scheduled from follow-on schedule check
